horovod/common/operations.cc

```diff
@@ -108,6 +110,10 @@ std::unique_ptr<OperationManager> op_manager;
 // Stall-check warning time
 #define STALL_WARNING_TIME std::chrono::seconds(60)
 
+// Stall-check shutdown time. If any rank is stalled for longer than this
+// horovod will shutdown
+#define STALL_SHUTDOWN_TIME std::chrono::seconds(180)
```

Comment:
This value is too small - we should make both warning time & shutdown time configurable through environment variables.  Default behavior for stall check should be to never timeout (for now).  Recoverable stall can happen if one of the ranks is doing full validation.  While not desirable, it should not be a fatal situation.

Reply:
That way, we should also be able to make an integration test with a small timeout value.

Reply:
In new diff this and stall warning time are configurable from environment variables.